processor/src/main/resources/org/mapstruct/ap/internal/model/source/builtin/XmlGregorianCalendarToLocalDateTime.ftl

```diff
@@ -0,0 +1,21 @@
+<#--
+
+    Copyright MapStruct Authors.
+
+    Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<#-- @ftlvariable name="" type="org.mapstruct.ap.internal.model.SupportingMappingMethod" -->
+private static <@includeModel object=findType("java.time.LocalDateTime")/> ${name}( <@includeModel object=findType("XMLGregorianCalendar")/> xcal ) {
+    if ( xcal == null ) {
+        return null;
+    }
+
+    return <@includeModel object=findType("java.time.LocalDateTime")/>.of(
```

Comment:
Perhaps we should check for undefined fields. Similar to what we are doing in `XmlGregorianCalendarToJodaLocalDateTime.ftl`.

We could also use the `nanoOfSecond` to create a time with the millis. For example to get the nano of seconds you could do `Duration.ofMillis( xcal.getMillisecond() ).getNano()`

Reply:
Thank @filiphr  for this, I have fixed and added some additional tests